### PR TITLE
Bedrock: fix new batcher, use sync status api, start from safe head

### PR DESCRIPTION
**Pull request by protolambda**
Previously the batcher would continue batch submitting from the L2 head.
However, if the sequencer runs  while the batch submitter is offline, the L2 chain continues, and the batch-submitter will be skipping a part. This causes a reorg, after which the batch submitter can't recover without restart (since it didn't detect reorgs).

This PR fixes the above, to make restarting of sequencer and batcher work without so much disruption of the L2 chain.

### CODE REVIEW

File: op-node/rollup/driver/state.go [hunk 1]
```diff
@@ -336,12 +348,35 @@ func (s *state) eventLoop() {
 				s.l2Head = unsafe
 				reqStep() // continue with the next step if we can
 			}
+		case respCh := <-s.syncStatusReq:
+			respCh <- SyncStatus{
+				CurrentL1:   s.derivation.Progress().Origin,
```
Comment (trianglesphere) [inline op-node/rollup/driver/state.go:353]: this is the tip of what we may consider while working on, realistically going to be the l1Head. Mainly a comment on the fact that "current L1" is not well defined (but then fine to define as is).

File: op-batcher/sequencer/driver.go [hunk 2]
```diff
@@ -5,6 +5,8 @@ import (
 	"math/big"
 	"time"
 
+	"github.com/ethereum-optimism/optimism/op-proposer/rollupclient"
```
Comment (trianglesphere) [inline op-batcher/sequencer/driver.go:8]: more weird goland import stuff.

**Review by trianglesphere — APPROVED** [review]
these changes look fine. Nice work marking the restarts way more efficient

File: op-node/rollup/driver/state.go [hunk 3]
```diff
@@ -13,6 +13,14 @@ import (
 	"github.com/ethereum/go-ethereum/log"
 )
 
+type SyncStatus struct {
```
Comment (tynes) [inline op-node/rollup/driver/state.go:16]: Having a godoc here would be nice

**Comment by protolambda** [comment]
Rebased onto `develop`, added `SyncStatus` go doc, and fixed import style.

**Pull request merged**